src/deb/deb-builder.mts

```diff
@@ -0,0 +1,273 @@
+
+		this.config.base.repo.forEach(channelEntry => {
+			debsPromises.push((async(): Promise<{ channel: string, debs: DebDescriptor[] }> => ({
+				channel: channelEntry.channel,
+				debs: await this.debsByPackages(channelEntry.packages),
+			}))());
+		});
+
+		const debs = await Promise.all(debsPromises);
+
+		debs.forEach(entry => {
+			this.debRepo.push(entry);
+		});
+	}
+
+	private async debsByPackages(packs: Packages): Promise<DebDescriptor[]> {
+		const debsPromises: Promise<DebDescriptor>[] = [];
+		const artsByBuildNumbersPromises: Promise<{version: string, artifacts: Artifact[]}>[] = [];
+
+		packs.packages.forEach(pack => {
+			artsByBuildNumbersPromises.push((async(): Promise<{ version: string, artifacts: Artifact[] }> => ({
+				version: pack.version,
+				artifacts: await this.artifactsProvider.artifactsByBuildNumber(pack.buildNumber),
+			}))());
+		});
+
+		const artsByBuildNumbers = await Promise.all(artsByBuildNumbersPromises);
+
+		artsByBuildNumbers.forEach(value => {
+			value.artifacts.filter(artifact => artifact.type === 'deb').forEach(artifact => {
+				debsPromises.push((async(): Promise<DebDescriptor> => ({
+					version: value.version,
+					artifact,
+					url: await this.artifactsProvider.artifactUrl(artifact),
+				}))());
+			});
+		});
+
+		return Promise.all(debsPromises);
+	}
+
+	private async dpkgScanpackages(): Promise<void> {
+		const promises: Promise<void>[] = [];
+
+		this.debRepo.forEach(channel => {
+			channel.debs.forEach(deb => {
+				promises.push(this.handleDeb(channel.channel, deb));
+			});
+		});
+
+		await Promise.all(promises);
+	}
+
+	private async handleDeb(channel: string, deb: DebDescriptor): Promise<void> {
+		const debUrl = deb.url;
+		const controlTarSizeRange = '120-129';
+
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+		const cocntrolTarSize = Number((await requestRange(debUrl, controlTarSizeRange)).read().toString()
+			.trim());
+		const controlTarRange = `132-${131 + cocntrolTarSize}`;
+		const controlTar = await requestRange(debUrl, controlTarRange);
+
+		const whereExtract = path.join(os.tmpdir(), `control-${crypto.randomBytes(4).toString('hex')}`);
+
+		createDir(whereExtract);
+
+		await new Promise<void>(resolve => {
+			controlTar
+				.pipe(tar.extract({ cwd: whereExtract, strip: 1 }, ['./control']))
+				// eslint-disable-next-line max-statements
+				.on('finish', () => {
+					const controlMetaContent = fs.readFileSync(path.join(whereExtract, 'control'), 'utf-8').replaceAll(':', '=');
+					const controlMeta = ini.parse(controlMetaContent);
+					const arch = controlMeta['Architecture'];
+
+					const archesSet = this.archesByChannel.get(channel);
+
+					if (archesSet) {
+						archesSet.add(arch);
+					} else {
+						this.archesByChannel.set(channel, new Set<string>([arch]));
+					}
+
+					const targetMetaPath = path.join(this.dists,
+						channel,
```

Comment:
Better call this `distribution` is Debian repo context. It is up to user configuration to decide what distributions correspond to "channels", like we call them in Windows/macOS repo.